mixer/adapter/stackdriver/helper/common.go

```diff
@@ -93,9 +95,14 @@ func (md *Metadata) FillProjectMetadata(in map[string]string) {
 func ToOpts(cfg *config.Params) (opts []gapiopts.ClientOption) {
 	switch cfg.Creds.(type) {
 	case *config.Params_ApiKey:
-		opts = append(opts, gapiopts.WithAPIKey(cfg.GetApiKey()))
+		if key := cfg.GetApiKey(); key != "" {
```

Comment:
shouldn't these be validation errors?

Reply:
You mean if an API key is set empty, it should be a validation error instead of defaulting something else?

Reply:
Yes. To me, validation should enforce that either it is `app_credentials` or a non-empty alternative, right?